app/src/main/java/com/codesoom/assignment/application/CatToyService.java

```diff
@@ -0,0 +1,83 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.CatToy;
+import com.codesoom.assignment.dto.CatToyModel;
+import com.codesoom.assignment.exception.CatToyNotFoundException;
+import com.codesoom.assignment.repository.CatToyRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+/**
+ * 고양이 장난감 생성, 조회, 수정, 삭제한다.
+ */
+@Service
+@Transactional
+public class CatToyService {
+
+    private final CatToyRepository catToyRepository;
+
+    public CatToyService(CatToyRepository catToyRepository) {
+        this.catToyRepository = catToyRepository;
+    }
+
+    /**
+     * 고양이 장난감을 생성한다.
+     * @param catToyModel
+     * @return 고양이 장난감
+     */
+    @ResponseStatus(HttpStatus.CREATED)
+    public CatToyModel createCatToy(CatToyModel catToyModel) {
+        CatToy catToy = new CatToy(catToyModel.name(), catToyModel.maker(), catToyModel.price(), catToyModel.imageUrl());
+        return new CatToyModel(catToyRepository.save(catToy));
+    }
+
+    /**
+     * 고양이 장난감 리스트를 조회하여 리턴한다.
+     * @return 고양이 장난감 리스트
+     */
+    public List<CatToyModel> selectCatToyList() {
+        return CatToyModel.ofList(catToyRepository.findAll());
+    }
+
+    /**
+     * 고양이 장난감 하나를 조회하여 리턴한다.
+     * @param id
+     * @return 고양이 장난감
+     */
+    public CatToyModel selectCatToy(Long id) {
+        return new CatToyModel(findCatToy(id));
+    }
+
+    /**
+     * 고양이 장난감을 수정한다.
+     * @param changeCatToyModel
+     * @return 고양이 장난감
+     */
+    public CatToyModel modifyCatToy(CatToyModel changeCatToyModel) {
+        CatToy catToy = findCatToy(changeCatToyModel.id());
+        catToy.changeCatToy(changeCatToyModel);
+        return new CatToyModel(catToyRepository.save(catToy));
+    }
+
+    /**
+     * 고양이 장난감을 삭제한다.
+     * @param catToyId
+     */
+    public void deleteCatToy(Long catToyId) {
+        catToyRepository.delete(findCatToy(catToyId));
+    }
+
+    /**
+     * 고양이 장난감이 있으면 조회, 없으면 CatToyNotFoundException 발생
+     * @param id
+     * @exception CatToyNotFoundException
+     * @return 고양이 장난감
+     */
```

Comment:
"예외"는 메인 디스크립션에 적을 정도의 메소드의 주요 기능이 아닙니다. 예외는 정말 예외적인 상황이죠. 

```suggestion
    /**
     * 고양이 장난감을 조회해 리턴합니다.
     * @param id 찾으려 하는 고양이 장난감의 아이디
     * @throws CatToyNotFoundException 고양이 장난감을 찾지 못한 경우
     * @return 고양이 장난감
     */
```